### PR TITLE
Fix a couple of longstanding zcashd build warnings

### PR DESCRIPTION
**Pull request by nuttycom**
Fixes zcash/zcash#2348

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -15,8 +15,6 @@ SUBDIRS += doc/man
 endif
 .PHONY: deploy FORCE rpc-tests
 
-GZIP_ENV="-9n"
```
Comment (str4d) [inline Makefile.am:18]: This was removed upstream in https://github.com/bitcoin/bitcoin/pull/11541, which was similarly fixing automake warnings. We should just backport the first commit from that PR (as it also removes some other code we similarly don't need).

**Review by str4d — CHANGES_REQUESTED** [review]
Backport request, and 3 questions.

File: src/secp256k1/configure.ac [hunk 2]
```diff
@@ -33,7 +33,7 @@ AM_PROG_CC_C_O
 # Restore AC_PROG_CC
 m4_rename_force([saved_AC_PROG_CC], [AC_PROG_CC])
 
-AC_PROG_CC_C99
+AC_PROG_CC
```
[the remark this answers is not in the document]
Comment (nuttycom) [reply src/secp256k1/configure.ac:36]: The build was reporting that `AC_PROG_CC_C99` is obsolete and recommended running `autoupdate`. That is also what introduced the `2.71` bump.

[the remark this answers is not in the document]
Comment (nuttycom) [reply src/secp256k1/configure.ac]: Reverted.

**Review by str4d — DISMISSED** [review]
utACK fc0621bfae34d43d310f6f347473e748a2979e57

**Review by daira — APPROVED** [review]
utACK

**Review by str4d — APPROVED** [review]
Re-utACK

**Review by daira — COMMENTED** [review]
Post-hoc re-ACK

**Pull request merged**